### PR TITLE
Allow user to set socket read timeout option

### PR DESCRIPTION
**Pull request by scottf**

### CODE REVIEW

File: src/main/java/io/nats/client/Options.java [hunk 1]
```diff
@@ -125,6 +125,11 @@ public class Options {
      */
     public static final long MINIMUM_SOCKET_WRITE_TIMEOUT_GT_CONNECTION_TIMEOUT = 100;
 
+    /**
+     * Constant used for calculating if a socket read timeout is large enough.
+     */
+    public static final long MINIMUM_SOCKET_READ_TIMEOUT_GT_CONNECTION_TIMEOUT = 100;
```
Comment (scottf) [inline src/main/java/io/nats/client/Options.java:131]: Same idea as write timeout buffer

File: src/main/java/io/nats/client/Options.java [hunk 2]
```diff
@@ -1273,6 +1286,16 @@ public Builder connectionTimeout(long connectionTimeoutMillis) {
...
+        /**
+         * Set the timeout to use around socket reads
+         * @param socketReadTimeoutMillis the timeout milliseconds
+         * @return the Builder for chaining
+         */
+        public Builder socketReadTimeoutMillis(int socketReadTimeoutMillis) {
+            this.socketReadTimeoutMillis = socketReadTimeoutMillis;
+            return this;
+        }
+
```
Comment (scottf) [inline src/main/java/io/nats/client/Options.java:1298]: I only take an int here because the Socket api only takes an int.

**Review by MauriceVanVeen — APPROVED** [review]
LGTM
Only one comment about a test failing due to a removed default value.

File: src/main/java/io/nats/client/Options.java [hunk 3]
```diff
@@ -840,7 +852,8 @@ public Builder properties(Properties props) {
             durationProperty(props, PROP_RECONNECT_JITTER_TLS, DEFAULT_RECONNECT_JITTER_TLS, d -> this.reconnectJitterTls = d);
             longProperty(props, PROP_RECONNECT_BUF_SIZE, DEFAULT_RECONNECT_BUF_SIZE, l -> this.reconnectBufferSize = l);
             durationProperty(props, PROP_CONNECTION_TIMEOUT, DEFAULT_CONNECTION_TIMEOUT, d -> this.connectionTimeout = d);
-            durationProperty(props, PROP_SOCKET_WRITE_TIMEOUT, DEFAULT_SOCKET_WRITE_TIMEOUT, d -> this.socketWriteTimeout = d);
+            intProperty(props, PROP_SOCKET_READ_TIMEOUT_MS, -1, i -> this.socketReadTimeoutMillis = i);
+            durationProperty(props, PROP_SOCKET_WRITE_TIMEOUT, null, d -> this.socketWriteTimeout = d);
```
[the remark this answers is not in the document]
Comment (scottf) [reply src/main/java/io/nats/client/Options.java:856]: yeah, fixing

**Review by MauriceVanVeen — APPROVED** [review]

**Pull request merged**